sqlglot/dialects/snowflake.py

```diff
@@ -740,6 +740,7 @@ class Tokenizer(tokens.Tokenizer):
             "TAG": TokenType.TAG,
             "TIMESTAMP_TZ": TokenType.TIMESTAMPTZ,
             "TOP": TokenType.TOP,
+            "WAREHOUSE": TokenType.WAREHOUSE,
```

Comment:
Can we double check if other dialects support `CREATE WAREHOUSE ...` too? If so, it'd make sense to have this in the base tokenizer instead.

Reply:
So I ran a prompt through Perplexity.ai / ChatGPT, and I checked GCP BigQuery (uses concept of slots), Databricks (no CREATE WAREHOUSE clause), AWS Redshift (basically PostgreSQL with some addenda), Azure Synapse Analytics (didn't find anything), Teradata (nope), and SAP SybaseIQ (nope). In addition I know the `CREATE WAREHOUSE` BNF grammar in Snowflake can become fairly specific, you can specify parameters for a given warehouse inline during warehouse creation.